### PR TITLE
Export index.js to be a function wrapped in closure that allows user …

### PR DESCRIPTION
**Pull request by philjung94**
…to set key token and then the Board ID via the graphql Query parameter

### CODE REVIEW

**Comment by philjung94** [comment]
Please review sir

File: graphql-library/src/trello.js [hunk 1]
```diff
@@ -1,85 +1,66 @@
-import 'dotenv/config'
 import https from 'https'
 const {TRELLO_OAUTH_TOKEN, TRELLO_API_KEY, TRELLO_BOARD_ID} = process.env
```
Comment (kevjy) [inline graphql-library/src/trello.js]: can remove this?

Comment (philjung94) [reply graphql-library/src/trello.js]: Why?

Comment (kevjy) [reply graphql-library/src/trello.js]: It's not used

File: graphql-library/src/trello.js [hunk 2]
```diff
@@ -37,17 +32,14 @@ export default class Trello {
    */
   withHandler(resolve, reject) {
     // @param {Object} res - Response object returned from https.get
-    if (![resolve, reject].every(f => typeof f === 'function')) {
-      throw new Error('resolve and reject must be functions, intended to be derived from a Promise executor.')
-    }
```
Comment (kevjy) [inline graphql-library/src/trello.js]: why remove this?

File: graphql-library/src/trello.js [hunk 3]
```diff
@@ -37,17 +32,14 @@ export default class Trello {
...
     return function callback(res) {
       const { statusCode } = res
       const contentType = res.headers['content-type']
       let err
-      if (statusCode !== 200) {
-        err = new Error('Request Failed.\n' + `Status Code: ${statusCode}`)
+      if (statusCode >= 400) {
```
Comment (kevjy) [inline graphql-library/src/trello.js]: Do we still need a catch-all for all status codes !== 200?

File: graphql-library/src/trello.js [hunk 4]
```diff
@@ -1,85 +1,58 @@
...
    * @param {String} query - the endpoint string without hostname, e.g. "/boards"
    */
-  return (query) => {
+  sanitizeEnds(query) {
     let qs = query
-    qs += `${qs.includes('?')? '&':'?'}key=${key}&token=${token}&card_limit=1000`
+    if(query.charAt(0) === '/') {
+      qs = query.slice(1)
+    }
+    if(['&','?'].some(char => char === qs.charAt(qs.length-1))) {
```
Comment (kevjy) [inline graphql-library/src/trello.js]:
```suggestion
    if (['&','?'].some(char => char === qs.charAt(qs.length-1))) {
```

File: graphql-library/src/trello.js [hunk 5]
```diff
@@ -1,85 +1,58 @@
...
-export function addAuth(key, token) {
   /**
+   * Parses the query string to ensure that excess chars are removed before passing it back to Trello.call
    * @param {String} query - the endpoint string without hostname, e.g. "/boards"
    */
-  return (query) => {
+  sanitizeEnds(query) {
     let qs = query
-    qs += `${qs.includes('?')? '&':'?'}key=${key}&token=${token}&card_limit=1000`
+    if(query.charAt(0) === '/') {
```
Comment (kevjy) [inline graphql-library/src/trello.js]:
```suggestion
    if (query.charAt(0) === '/') {
```

File: graphql-library/src/index.js [hunk 6]
```diff
@@ -1,23 +1,38 @@
+/*
+ * Library Example Usage
+ * =====================
+ * import 'dotenv/config'
+ * import {trelloKey, trelloToken} from process.env
+ * const myQuery = `query {...MySDL}`
+ * const ltot = new LTOT(trelloKey, trelloToken)
+ * ltot.query(myQuery).then(console.log) // returns a promise.
+ */
```
Comment (kevjy) [inline graphql-library/src/index.js]: Nice inclusion of example usage! Would also be good to include the import line for `LTOT`?

File: graphql-library/src/trello.js (hunk 3 above)
Comment (philjung94) [reply graphql-library/src/trello.js]: Yes

File: graphql-library/src/index.js [hunk 7]
```diff
@@ -3,37 +3,37 @@
  * =====================
  * import 'dotenv/config'
  * import {trelloKey, trelloToken} from process.env
- * import LTOT from 'ltot-gql'
+ * import LT from 'living-trello'
```
Comment (kevjy) [inline graphql-library/src/index.js]: also update the name in `package.json`?

**Pull request merged**